manifests/0000_70_cluster-network-operator_01_egr_crd.yaml

```diff
@@ -0,0 +1,128 @@
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: EgressRouterSpec contains the configuration for an egress
+              router.
+            properties:
+              ip:
+                description: IP configuration arguments.
+                properties:
+                  addresses:
+                    description: List of IP addresses to configure on the interface.
+                    items:
+                      description: EgressRouterAddress contains a pair of ip cidr
+                        and gateway to be configured on the router's interface
+                      properties:
+                        gateway:
+                          description: IP address of the next-hop gateway, if it cannot
+                            be automatically determined. Can be IPv4 or IPv6.
+                          type: string
+                        ip:
+                          description: IP is the address to configure on the router's
+                            interface. Can be IPv4 or IPv6.
+                          type: string
+                      required:
+                      - ip
+                      type: object
+                    type: array
+                  allowedDestinations:
+                    description: List of CIDR blocks that the pod is allowed to connect
+                      to via this interface. If not provided, the pod can connect
+                      to any destination.
+                    items:
+                      description: CIDR represents an IPv4 or IPV6 CIDR block.
+                      type: string
+                    type: array
+                required:
+                - addresses
+                type: object
+              networkInterface:
+                description: Specification of interface to create/use. e.g; macvlan
+                properties:
+                  macVLAN:
+                    description: Arguments specific to the interfaceType macvlan
+                    properties:
+                      master:
+                        description: Name of the master inteface. Usually used for
+                          macvlan and ipvlan. Need not be specified if it can be inferred
+                          from the IP address.
+                        minLength: 1
+                        type: string
+                      mode:
+                        description: Usually used for macvlan interface.
```

Comment:
Since mode doesn't take any random value, we should document explicitly which are the values allowed. It's extremely annoying when oc explain doesn't do this. Is there a default value?

Also it would be great if we could add a pattern here to make sure we only allow valid values.